### PR TITLE
Allow HEAPX symbols in EXPORT_RUNTIME_METHODS

### PR DESCRIPTION
**Pull request by sbc100**
- Factor out updateMemoryViews.
- Enable HEAPXX exports to be specified in EXPORT_RUNTIME_METHODS.
- Add all HEAP symbosl to EXPORT_RUNTIME_METHODS unless in STRICT mode.

Replaces #21407

### CODE REVIEW

File: test/other/metadce/test_metadce_hello_O0.jssize [hunk 1]
```diff
@@ -1 +1 @@
-22913
+23002
```
Comment (brendandahl) [inline test/other/metadce/test_metadce_hello_O0.jssize:1]: It looks like js size when up in a few of these. Any idea why?

File: src/preamble_minimal.js [hunk 2]
```diff
@@ -116,6 +85,10 @@ else {
 #endif // MODULARIZE
 #endif // PTHREADS
 
+#if ASSERTIONS && SHARED_MEMORY
```
Comment (brendandahl) [inline src/preamble_minimal.js:88]: Unrelated change?

File: src/modules.js [hunk 3]
```diff
@@ -349,6 +349,10 @@ function exportRuntime() {
   // being exported. how we show the message depends on whether it's
   // a function (almost all of them) or a number.
   function maybeExport(name) {
+    // HEAP objects are exported separetly in updateMemoryViews
```
Comment (brendandahl) [inline src/modules.js:352]: s/separetly/separately

File: test/other/metadce/test_metadce_hello_O0.jssize (hunk 1 above)
Comment (sbc100) [reply test/other/metadce/test_metadce_hello_O0.jssize:1]: These are `-O0` tests in which `ASSERTSIONS` are enabled.  When `ASSERTIONS` are enabled we inject stubs to detect the use of unexported runtime functions.   This means that whenever we add new runtime function the `ASSETIONS` builds get a little bigger.  This is expected and not a problem (IMHO)

File: src/preamble_minimal.js (hunk 2 above)
Comment (sbc100) [reply src/preamble_minimal.js:88]: Actually this is an assert from the `MINIMAL_RUNTIME` version of the `updateMemoryViews` function.  I'm not sure we really need to preserve it but I decided to keep it around to be on the safe side.

**Review by brendandahl — APPROVED** [review]
